modules/light-clients/14-avalanche/client_state.go

```diff
@@ -0,0 +1,290 @@
+package avalanche
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v7/modules/core/24-host"
+
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+var _ exported.ClientState = (*ClientState)(nil)
+
+const (
+	// MaxChainIDLen is a maximum length of the chain ID.
+	MaxChainIDLen = 50
+)
+
+// NewClientState creates a new ClientState instance
+func NewClientState(
+	chainID string, trustLevel Fraction,
+	trustingPeriod time.Duration,
+	latestHeight clienttypes.Height,
+	upgradePath []string,
+) *ClientState {
+	return &ClientState{
+		ChainId:        chainID,
+		TrustLevel:     trustLevel,
+		TrustingPeriod: trustingPeriod,
+		LatestHeight:   latestHeight,
+		FrozenHeight:   clienttypes.ZeroHeight(),
+		UpgradePath:    upgradePath,
+	}
+}
+
+func (cs *ClientState) ClientType() string {
+	return exported.Avalanche
+}
+
+func (cs *ClientState) GetLatestHeight() exported.Height {
+	return cs.LatestHeight
+}
+
+func (cs *ClientState) Validate() error {
```

Comment:
Looks good, there is no `ProofSpecs` definition like there is for the tendermint light client but I assume this is because of the different method of verifying the merkle proof (discussion on eth_getProof).